scipy/spatial/transform/rotation.py

```diff
@@ -1463,6 +1464,73 @@ def magnitude(self):
         else:
             return angles
 
+    def reduce(self, left=None, right=None, return_best=False):
+        """Reduce this rotation with the provided rotation groups.
+
```

Comment:
Explanation is required.

Reply:
It is quite hard to come up with a good description. Do you have a proposal?

Reply:
OK, I'll try later.

Reply:
Reduction of a rotation ``p`` is a transformation of the form ``q = l * p * r``, where ``l`` and ``r`` are chosen from `left` and `right` respectively, such that rotation ``q`` has the smallest magnitude. 

If `left` and `right` are rotation groups representing symmetries of two objects rotated by ``p``, then ``q`` is the rotation of the smallest magnitude to align these objects considering their symmetries.

Reply:
Maybe we also should add explanation, that when `left` and `right` are the same **groups**, you don't need to reduce by both (which follows from the math)

Reply:
Thanks for this.  The description is great.

>Maybe we also should add explanation, that when left and right are the same groups, you don't need to reduce by both (which follows from the math)

I looked at this again.  The reduced rotations have the same magnitude (which is useful for calculating disorientation angles) but are otherwise different.  Here is a code stub which shows this:
```
g = Rotation.create_group("O")
r = Rotation.random(10)

f = r.reduce(left=g, right=g)
fl = r.reduce(left=g)
fr = r.reduce(right=g)

print(f.as_quat())
print(fl.as_quat())
print(fr.as_quat())
```

Reply:
> The reduced rotations have the same magnitude (which is useful for calculating disorientation angles) but are otherwise different.

Yes, I believe there is no single rotation even in a group to reproduce `l * p * r`, but there is a rotation `r * l.inv()` to reproduce the magnitude.